pkg/entities/message.go

```diff
@@ -45,3 +55,40 @@ func (m *MsgBuffer) GetDataRecFlag() bool {
 func (m *MsgBuffer) SetDataRecFlag(flag bool) {
 	m.dataRecFlag = flag
 }
+
+type templateMessage struct {
```

Comment:
can this abstraction be moved as templateSet and dataSet? As message can have both data and templates? In addition, maybe its better to use decodedTemplateSet and decodedDataSet to differentiate.